enterprise/backend/src/metabase_enterprise/semantic_search/pgvector_api.clj

```diff
@@ -79,11 +81,50 @@
   "Indexes documents into the active semantic search index.
   Documents are upserted - existing documents with same id are replaced.
 
+  This is the 'immediate mode' API for inserting documents, the expected production behaviour
+  is to instead (gate-updates!) the documents and allow the indexer task to pick them up.
+
   `documents` is a logical collection, but can be reducible to save memory usage."
   [pgvector index-metadata documents]
   (let [{:keys [index]} (ensure-active-index-state pgvector index-metadata)]
     (semantic.index/upsert-index! pgvector index documents)))
 
+(defn gate-updates!
```

Comment:
It'd prefer a single `gate!` function rather han seperate update/delete functions (simple proxy, maybe only adding h the partitioning behaviour).  This would be both a richer and simpler calling pattern IMO regarding the interleaved update/delete gate! actually supports - but it would involve hoisting up the gate doc mapping. Maybe in the future.